### PR TITLE
add latest sol2 v2 for having a c++14 version

### PR DESCRIPTION
**Pull request by a4z**
sol2 v3 requires C++17, so for people with older compilers
this might be useful

Specify library name and version:  **sol2/2.20.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sol2/config.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 versions:
-  3.0.3:
-    folder: 3.x.x
-
```
Comment (a4z) [inline recipes/sol2/config.yml:4]: please tell ,e what the gab between line 4 and 5 on the right hand side means, is it the reason for all the problems ?

File: recipes/sol2/config.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 versions:
-  3.0.3:
-    folder: 3.x.x
-    
\ No newline at end of file
+  "2.20.6":
+    folder: "2.x.x"
+  "3.0.3":
+    folder: "3.x.x"
```
Comment (a4z) [inline recipes/sol2/config.yml:5]: why do I see a gap between line 4 and 5 and is this the reason for all the problems?

Comment (Croydon) [reply recipes/sol2/config.yml:5]: No clue if this is related, but please make sure to configure your editor(s) that they add newlines at the end of text files. Missing new lines can cause issues

**Comment by danimtb** [comment]
Sorry @a4z, think this PR is affected by https://github.com/conan-io/conan-center-index/issues/1386 we will try to fix it soon

**Comment by a4z** [comment]
> Sorry @a4z, think this PR is affected by #1386 we will try to fix it soon

Thanks! Now I know

**Comment by jgsogo** [comment]
Issue https://github.com/conan-io/conan-center-index/issues/1386 should be fixed. PR #1312 is already green, this should be too... but we have some conflicts here to solve first

Thanks for your patience

File: recipes/sol2/2.x.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        minimal_cpp_standard = "14"
+        try:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        except ConanInvalidConfiguration:
+            raise
+        except ConanException:
+            # FIXME: We need to handle the case when Conan doesn't know
+            # about a user defined compiler's default standard version
+            self.output.warn(
+                "Unnable to determine the default standard version of the compiler")
```
Comment (uilianries) [inline recipes/sol2/2.x.x/conanfile.py:46]:
```suggestion
    def configure(self):
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, "14")
        compiler = str(self.settings.compiler)
        comp_version = tools.Version(self.settings.compiler.version)
        compilers = {"Visual Studio": "14", "gcc": "5", "clang": "3.2", "apple-clang": "4.3"}
        min_version = compilers.get(compiler)
        if not min_version :
            self.output.warn("sol2 recipe lacks information about the %s compiler support".format( compiler))
        elif comp_version < min_version:
            raise ConanInvalidConfiguration("sol2 requires C++14 or higher support standard."
                                            " {} {} is not supported."
                                            .format(compiler, comp_version))
```

Comment (a4z) [reply recipes/sol2/2.x.x/conanfile.py:46]: Thanks!

**Pull request merged**